src/mesher/Mesher.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include "gmsh.h"
+#include <string>
+#include "TopoDS_Shape.hxx"
+
+
+class Mesher 
+{
+public: 
+
+    class Options 
+    {
+    public:
+
+    void setDimension(int dimension);
+
+    int getDimension();
+
+    void setSize(int mySize);
+
+    int getSize();
+
+    void setName(std::string myName);
+
+    std::string getName();
+ 
+    private:
+        int dim;
+        double size;
+        std::string fileName; 
+    };
```

Comment:
Since you have getters and setters for all data members, there is no real point in having the data members private. The code would be much shorter and easier to maintain if you just defined a struct with public fields:

```cpp
struct Options {
    int dim;
    double size;
    std::string fileName;
};
```

The option name `size` coul probably be improved: What is the size of a mesh? How is this option used/how does this option influence the generated mesh? Can we find a name that conveys its meaning better?